src/sdk/ConversationTranslator/ConversationTranslator.ts

```diff
@@ -43,7 +82,111 @@ export class ConversationTranslator implements IConversationTranslator, IDisposa
      * @param err
      */
     public joinConversationAsync(conversation: any, nickname: any, lang?: any, cb?: () => void, err?: (e: string) => void): void {
-        throw new Error("Method not implemented.");
+
+        try {
+            Contracts.throwIfNullOrUndefined(conversation, "conversation");
+            Contracts.throwIfNullOrUndefined(nickname, "nickname");
+
+            if (this.privConversation) {
+                this.privConversation.close();
+                this.privConversation = undefined;
+                this.privSpeechTranslationConfig.close();
+                this.privSpeechTranslationConfig = undefined;
+            }
+
+            if (typeof conversation === "string") {
+
+                if (lang === undefined) { lang = ConversationTranslatorConfig.defaultLanguageCode; }
+
+                // create a placecholder config
+                this.privSpeechTranslationConfig = SpeechTranslationConfig.fromSubscription("abcdefghijklmnopqrstuvwxyz012345", "westus");
+                this.privSpeechTranslationConfig.setProfanity(ProfanityOption.Masked);
+                this.privSpeechTranslationConfig.addTargetLanguage(lang);
+                this.privSpeechTranslationConfig.setProperty(PropertyId[PropertyId.SpeechServiceConnection_RecoLanguage], lang);
+
+                this.privSpeechTranslationConfig.setProperty(PropertyId[PropertyId.ConversationTranslator_Name], nickname);
+
+                const endpoint: string = this.privProperties.getProperty(PropertyId.ConversationTranslator_Host);
+                if (endpoint) {
+                    this.privSpeechTranslationConfig.setProperty(PropertyId[PropertyId.ConversationTranslator_Host], endpoint);
+                }
+                const speechEndpointHost: string = this.privProperties.getProperty(PropertyId.ConversationTranslator_SpeechHost);
```

Comment:
Please use the existing PropertyId::SpeechServiceConnection_Endpoint instead

Reply:
I've updated it to 'SpeechServiceConnection_Host'.